### ISSUE TITLE
Move internal implementation from Zend\Code to PHP-Parser

### ISSUE THREAD

**Issue by Ocramius** [Ocramius/ProxyManager#49]
Moving to PHP-Parser will allow inlining of parent class code, therefore reducing linear overhead and allowing more performance optimizations.

**Comment by robfrawley** [comment]
:+1:  omg, yes - please! Need help? ;-)

**Comment by malukenho** [comment]
@robfrawley fell free to send PRs :smile:

**Comment by Ocramius** [comment]
I'm not sure if this is interesting/viable anymore.

There is no clear advantage in using PHP-Parser unless very obvious automated optimizations can be applied.

**Comment by Ocramius** [comment]
Note: can easily be fixed if Roave/BetterReflection#39 is carefully checked

**Comment by lisachenko** [comment]
I also working on parser-reflection, working concept is ready. Main feature: all classes will extend original ones, so all existing libraries can work with them. Initialization will be compatible too: `new ParserReflection\ReflectionClass(SomeClass::class)`

**Comment by Ocramius** [comment]
Yeap, BetterReflection does that as well, but @asgrim is pretty much
working on it full time, so I suggest joining him on it ;-)

Marco Pivetta

http://twitter.com/Ocramius

http://ocramius.github.com/

**Comment by lisachenko** [comment]
> Yeap, BetterReflection does that as well, but @asgrim is pretty much

No, according to the https://github.com/Roave/BetterReflection/blob/master/src/Reflection/ReflectionClass.php#L11, his classes don't extend original ones, just implements an interface which looks like reflection, so I can't accept this solution for my needs. I need to reflect code transparently, like traditional reflection classes do.

**Comment by asgrim** [comment]
@lisachenko I'm planning on creating an adapter to make this possible, see Roave/BetterReflection#18
